play.pokemonshowdown.com/src/battle-tooltips.ts

```diff
@@ -1514,26 +1538,93 @@ export class BattleTooltips {
+				'Glalitite': { 'sp': 20, 'skip': ['Glalie-Mega'] },
+				'Gyaradosite': { 'sp': 0, 'skip': ['Gyarados-Mega'] },
+				'Heracronite': { 'sp': -10, 'skip': ['Heracross-Mega'] },
+				'Houndoominite': { 'sp': 20, 'skip': ['Houndoom-Mega'] },
+				'Kangaskhanite': { 'sp': 20, 'skip': ['Kangaskhan-Mega'] },
+				'Latiasite': { 'sp': 0, 'skip': ['Latias-Mega'] },
+				'Latiosite': { 'sp': 0, 'skip': ['Latios-Mega'] },
+				'Lopunnite': { 'sp': 30, 'skip': ['Lopunny-Mega'] },
+				'Lucarionite': { 'sp': 22, 'skip': ['Lucario-Mega'] },
+				'Manectite': { 'sp': 30, 'skip': ['Manectric-Mega'] },
+				'Mawilite': { 'sp': 0, 'skip': ['Mawile-Mega'] },
+				'Medichamite': { 'sp': 20, 'skip': ['Medicham-Mega'] },
+				'Metagrossite': { 'sp': 40, 'skip': ['Metagross-Mega'] },
+				'Mewtwonite X': { 'sp': 0, 'skip': ['Mewtwo-Mega-X'] },
+				'Mewtwonite Y': { 'sp': 10, 'skip': ['Mewtwo-Mega-Y'] },
+				'Pidgeotite': { 'sp': 20, 'skip': ['Pidgeot-Mega'] },
+				'Pinsirite': { 'sp': 20, 'skip': ['Pinsir-Mega'] },
+				'Red Orb': { 'sp': 0, 'skip': ['Groudon-Primal'] },
+				'Sablenite': { 'sp': -30, 'skip': ['Sableye-Mega'] },
+				'Salamencite': { 'sp': 20, 'skip': ['Salamence-Mega'] },
+				'Sceptilite': { 'sp': 25, 'skip': ['Sceptile-Mega'] },
+				'Scizorite': { 'sp': 10, 'skip': ['Scizor-Mega'] },
+				'Sharpedonite': { 'sp': 10, 'skip': ['Sharpedo-Mega'] },
+				'Slowbronite': { 'sp': 0, 'skip': ['Slowbro-Mega'] },
+				'Steelixite': { 'sp': 0, 'skip': ['Steelix-Mega'] },
+				'Swampertite': { 'sp': 10, 'skip': ['Swampert-Mega'] },
+				'Tyranitarite': { 'sp': 10, 'skip': ['Tyranitar-Mega'] },
+				'Venusaurite': { 'sp': 0, 'skip': ['Venusaur-Mega'] },
+				'Rusted Sword': { 'sp': 10, 'skip': ['Zacian-Crowned'] },
+				'Rusted Shield': { 'sp': -10, 'skip': ['Zamazenta-Crowned'] },
+				'Adamant Crystal': { 'sp': 0, 'skip': ['Dialga-Origin'] },
+				'Griseous Core': { 'sp': 0, 'skip': ['Giratina-Origin'] },
+				'Lustrous Globe': { 'sp': 20, 'skip': ['Palkia-Origin'] },
+				'Vile Vial': { 'sp': 37, 'skip': ['Venomicon-Epilogue'] },
+				'Hearthflame Mask': { 'sp': 0, 'skip': ['Ogerpon-Hearthflame', 'Ogerpon-Hearthflame-Tera'] },
+				'Wellspring Mask': { 'sp': 0, 'skip': ['Ogerpon-Wellspring', 'Ogerpon-Wellspring-Tera'] },
+				'Cornerstone Mask': { 'sp': 0, 'skip': ['Ogerpon-Cornerstone', 'Ogerpon-Cornerstone-Tera'] },
+			};
+			const stone = pokemon.item as 'Abomasite';
+			if (!(megaDelta[stone]?.skip.includes(pokemon.speciesForme))) {
+				baseSpe += megaDelta[stone]?.sp || 0;
+				if (baseSpe < 1) baseSpe = 1;
+				if (baseSpe > 255) baseSpe = 255;
+
+			}
 		}
-		if (rules['350 Cup Mod'] && species.bst <= 350) {
-			baseSpe *= 2;
+		if (rules['Tier Shift Mod'] && this.battle.gen === 9) {
+			const boosts = { 'uu': 15, 'rubl': 15, 'ru': 20, 'nubl': 20, 'nu': 25,
+				'publ': 25, 'pu': 30, 'zubl': 30, 'zu': 30, 'nfe': 30, 'lc': 30 };
+			baseSpe += boosts[species.tier.toLowerCase() as 'uu'] ?? 0;
```

Comment:
if you addressed my suggestion to add a type to this you wouldn't need to typecast the species tier as `uu`. this also doesn't account for parenthetical tiers, so it's better to just get the ID of the tier.

Reply:
hopefully done

Reply:
pu by technicality seems to no longer be relevant, ou as a technicality exists and should just be grabbed by the coalesence to 0